java/src/main/java/io/trino/velox/execution/NativeSqlTaskExecutionManager.java

```diff
@@ -195,11 +197,13 @@ public void fetchOutputFromNative(String id, int partitionId)
                     if (!output.isEmpty()) {
                         taskExecution.updateOutput(partitionId, output);
                         // Register next listener.
-                        trinoBridge.registerOutputPartitionListener(nativeHandler,
-                                taskId.toString(),
-                                partitionId,
-                                taskExecution.getPartitionSequenceId(partitionId),
-                                maxOutputPageFetchBytes);
+                        taskExecution.isOutputBufferFull().addListener(() -> {
```

Comment:
If the output buffer is full, it will maintain partition output data on the `PartitionedOutputBufferManager`, which will cause memory usage on the native side while reducing the memory usage on the Java side. But for total memory usage, only if there is a back-pressure mechanism on the native side, it is useless. So maybe we don't need to add this listener?

Reply:
```PartitionedOutputBufferManager``` also has a memory limitation in ```enqueue```
```c++
    if (totalSize_ > maxSize_ && future) {
      promises_.emplace_back("PartitionedOutputBuffer::enqueue");
      *future = promises_.back().getSemiFuture();
      blocked = true;
    }
```
This will block ```PartitionOutputOperator```.

Reply:
If the Java side output buffer is full, and exchange is blocked because of memory limitation in `enqueue`, I think there might block the execution?

Reply:
Yes, the output pipeline will be blocked until the listener is registered and invoked.

Reply:
So there is a corner case that if a query contains 2 tasks each contains 1 pipeline. One called A is `TableScan --> PartitionedOutput`, another one called B is `Exchange ---> PartitionedOutput`(I am not sure if there will be a pipeline like this), and then, A is blocked by the full output buffer on the Java side and B is blocked by the enqueuing in PartitionedOutput. The Java side memory will not be released because the blocking in PartitionedOutput caused the blocking in Exchange. So the query will never finish. Will this corner case happen?

Reply:
In this case, I think the query is driven by the consumer of B (e.g., downstream tasks or Trino CLI). As the consumer pulls or cancels results from B, the tasks will be activated or aborted.

Reply:
Yes, you are right. I prefer fast fail in this case, but it's okay.